app/models/supermarket.rb

```diff
@@ -1,3 +1,7 @@
 class Supermarket < ApplicationRecord
   has_many :items
```

Comment:
Something that would make the join a lot easier to find unique customers: set up the Supermarket model to take advantage of the associations that items, its child object has. 
Similar to how you use it with many-to-many relationships, you could add the following association to the Supermarket model:
`has_many :customers, through: :items`

Under the hood, AR will build an association that gets all the customers associated with the items that belong to this store. This is a multi-step join under the hood, but with AR, we can just call `supermarket.customers` and get the results!